src/flake8_tidy_imports/checks.rs

```diff
@@ -22,3 +26,71 @@ pub fn banned_relative_import(
         None
     }
 }
+
+pub fn name_is_banned<T>(
+    located: &Located<T>,
+    name: String,
+    banned_apis: &FxHashMap<String, BannedApi>,
+) -> Option<Check> {
+    if let Some(ban) = banned_apis.get(&name) {
+        return Some(Check::new(
+            CheckKind::BannedApi {
+                name,
+                message: ban.msg.to_string(),
+                attribute_access: false,
+            },
+            Range::from_located(located),
+        ));
+    }
+    None
+}
+
+pub fn name_or_parent_is_banned<T>(
+    located: &Located<T>,
+    mut name: &str,
+    banned_apis: &FxHashMap<String, BannedApi>,
+) -> Option<Check> {
+    loop {
+        if let Some(ban) = banned_apis.get(name) {
+            return Some(Check::new(
+                CheckKind::BannedApi {
+                    name: name.to_string(),
+                    message: ban.msg.to_string(),
+                    attribute_access: false,
+                },
+                Range::from_located(located),
+            ));
+        }
+
+        match name.rfind('.') {
+            Some(idx) => {
+                name = &name[..idx];
+            }
+            None => break,
+        }
+    }
+    None
+}
+
+pub fn banned_attribute_access(
+    checker: &mut Checker,
```

Comment:
Typically, if methods take `checker`, then we just call `checker.add_check` rather than returning an `Option<Check>`.